src/iblt.h

```diff
@@ -40,21 +40,50 @@ SOFTWARE.
 // Mitzenmacher
 //
 
-class HashTableEntry
+const uint64_t IBLT_MAX_VERSION_SUPPORTED = 2;
+
+class BaseHashTableEntry
 {
 public:
     int32_t count;
     uint64_t keySum;
     uint32_t keyCheck;
     std::vector<uint8_t> valueSum;
 
-    HashTableEntry() : count(0), keySum(0), keyCheck(0) {}
-    bool isPure() const;
+    BaseHashTableEntry() : count(0), keySum(0), keyCheck(0) {}
+    bool isPure(uint32_t keycheckMask) const;
     bool empty() const;
     void addValue(const std::vector<uint8_t> &v);
+};
+
+class HashTableEntry : public BaseHashTableEntry
+{
+public:
+    uint64_t keyCheck64;
 
     ADD_SERIALIZE_METHODS;
 
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream &s, Operation ser_action)
+    {
+        READWRITE(count);
+        READWRITE(keySum);
+
+        if (!ser_action.ForRead())
+            keyCheck64 = (uint64_t)keyCheck;
+        READWRITE(COMPACTSIZE(keyCheck64));
```

Comment:
As per discussion on keybase - maybe use a global way to squeeze the bits together here (as they are all the same for the whole IBLT), instead of COMPACTSIZE encoding?

Reply:
I'm not a big fan of this approach, which is the way that I [originally](https://github.com/bissias/BitcoinUnlimited/pull/19) implemented variable keysumCheck. Squashing the bits requires that they be pulled out of HashTableEntry and into CIblt. As a result, nearly every method signature in HashTableEntry changes, and as a result, there are small changes to nearly every method body in CIblt. The only way I could see to keep things clean while maintaining backward compatibility was to implement two separate versions of HashTableEntry and CIblt. It is also required to keep two different pointers in CGrapheneSet (depending on which CIblt implementation is being used), although only one need be serialized. Given the messiness, and my admittedly cursory estimate (see link above) that IBLTs will comprise no more than 10% of large graphene blocks, it seems like a lot of tech debt to add for not a whole lot of space savings. That said, I'm open to further discussion.